### PR TITLE
fix: include JSX related module in package

### LINKED ISSUE #503: JSX related files excluded from package
Should be added here:

https://github.com/snabbdom/snabbdom/blob/fa7b7d0796cfe2e8a0e61596beb4cbebeb53a146/package.json#L80-L235

### PR DESCRIPTION
**Pull request by mightyiam**
Fixes #503.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -241,6 +247,10 @@
     "/is.d.ts",
     "/is.js",
     "/is.js.map",
+    "/jsx-global.d.ts",
+    "/jsx.d.ts",
+    "/jsx.js",
+    "/jsx.js.map",
```
Comment (mightyiam) [inline package.json]: Package the CommonJS version of the JSX related modules.

File: src/jsx-global.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ declare global {
    * opt-in jsx intrinsic global interfaces
    * see: https://www.typescriptlang.org/docs/handbook/jsx.html#type-checking
    */
+  // eslint-disable-next-line @typescript-eslint/no-namespace
```
Comment (mightyiam) [inline src/jsx-global.ts]: In a `.d.ts` file it doesn't complain. Perhaps we are not linting those at all? In any case, just ignoring this error seems to be the right thing to do.

File: tsconfig.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
         "src/htmldomapi.ts",
         "src/hooks.ts",
         "src/is.ts",
-        "src/jsx-global.d.ts",
+        "src/jsx-global.ts",
```
Comment (mightyiam) [inline tsconfig.json]: This source file is renamed.

File: src/jsx-global.ts (hunk 2 above)
Comment (nojvek) [reply src/jsx-global.ts]: I'm guessing namespaces might be allowed in .d.ts but not in .ts

**Comment by nojvek** [comment]
Will fix #502 to just have sel shorthand once you merge this.

**Comment by mightyiam** [comment]
Thank you, @nojvek.

**Pull request merged**